File: 1019-squares-of-a-sorted-array/1019-squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int len = nums.length;
        int res[] = new int[len];
        int left = 0, right = len-1, num=0;
        int i=right;
        //for (int i=right; i>=0; i--){
        while (left<=right){            
              if (Math.abs(nums[left]) < 
                    Math.abs(nums[right]) ){
                num = nums[right];
                right--;                  
              }else{
                num = nums[left];
                left++;
              }
            //res[i] = num * num;
            res[i--] = num * num;            
        }
        return res;
    }
}

File: 1019-squares-of-a-sorted-array/1019-squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        arr_len = len(nums)
        newarr = [0] * arr_len
        begin = 0
        end = arr_len-1
        pos = end
        while begin <= end:
            if abs(nums[begin]) < abs(nums[end]):
                newarr[pos] = nums[end] * nums[end]
                end-=1
                pos-=1
            else:
                newarr[pos] = nums[begin] * nums[begin]
                begin+=1
                pos-=1

        return newarr
